docs/source/backends.rst

```diff
@@ -28,21 +28,30 @@ The following is a brief overview of libraries which have been tested with
     - `jax <https://github.com/google/jax>`_: compiled GPU tensor expressions
       including ``autograd``-like functionality
 
+``opt_einsum`` is agnostic to the type of n-dimensional arrays (tensors)
+it uses, since finding the contraction path only relies on getting the shape
+attribute of each array supplied.
+It can perform the underlying tensor contractions with various
+libraries. In fact, any library that provides a :func:`~numpy.tensordot` and
+:func:`~numpy.transpose` implementation can perform most normal contractions.
+While more special functionality such as axes reduction is reliant on a
+:func:`~numpy.einsum` implementation.
+
 .. note::
 
     For a contraction to be possible without using a backend einsum, it must
-    satisfy the following rule: in the full expression (so *including* output
+    satisfy the following rule: in the full expression (*including* output
     indices) each index must appear twice. In other words, each dimension
     must be contracted with one other dimension, or left alone.
 
 
-General backend for any ndarray
-===============================
+Backend agnostic contractions
+=============================
 
-This 'duck-typing' support just requires specifying the correct ``backend``
+The automatic backend detection will be detected based on the first supplied
```

Comment:
You could say "The automatic backend detection dispatches on the first supplied array..."" just to avoid detect twice (and it is essentially a single-dispatch system).